### ISSUE TITLE
Invite to a room that has been deleted by a server admin cannot be accepted nor rejected

### ISSUE THREAD

**Issue by eras** [vector-im/element-meta#1010]
### Steps to reproduce

1. Get invited to a room
2. Somehow arrange the room to be removed (probably by server admins)
3. Try to reject the invite

#### What did you expect?
I expected the invite to get rejected and the invite to get removed from my Element.

#### What happened instead?
The invite sticks permanently until clearing caches. A dialog pops up indicating the endpoint had returned a HTTP error (lost the logs already so I don't have the specific HTTP code).

Trying any of the three options (accept, reject, reject & ignore) available has the same outcome.

### Operating system

Linux

### Browser information

Firefox

### Application version

1.11.17

### Homeserver

matrix.org

### Will you send logs?

No

Related issues: vector-im/element-meta#1000 vector-im/element-meta#1635

**Comment by turt2live** [comment]
This sounds like a Synapse bug, as this used to work.

**Comment by t3chguy** [comment]
Yeah this is an issue where Synapse doesn't send it down /sync coming back

**Comment by richvdh** [comment]
> This sounds like a Synapse bug, as this used to work.

I'm not sure this ever worked. You may be thinking of https://github.com/matrix-org/synapse/issues/2181, but in that case it was a *remote* invite from a server that became unreachable. In this case, the invite is from a user on the *local* server, but the room has since been deleted, which means we cannot generate a "leave" event.

**Comment by richvdh** [comment]
This is also reported for element-android at https://github.com/vector-im/element-android/issues/4720.

**Comment by richvdh** [comment]
I'll file a new bug for Synapse.

**Comment by richvdh** [comment]
https://github.com/matrix-org/synapse/issues/15085

**Comment by richvdh** [comment]
Workaround for this is to clear the client-side cache, by the way.

**Comment by Elliria** [comment]
Mine finally went away after a couple of days. The issue had nothing to do with cache or cookies for me, because my browser is set to clear those every time it's closed.

**Comment by Elliria** [comment]
I have the same issue on gitter.im in the Element browser client.

**Comment by Elliria** [comment]
I spoke too soon. The invitation went away, but the room still exists and exhibits the same issue when I try to join the conversation in it.

**Comment by richvdh** [comment]
@Elliria I think you have a different issue.

**Comment by Elliria** [comment]
Okay.
